src/NServiceBus.Core.Analyzer/Sagas/RewriteConfigureHowToFindSagaFixer.cs

```diff
@@ -0,0 +1,194 @@
+
+                    return new SagaMessageMapping(0, messageType, null, isHeader, true, mappingExpression, null);
+                });
+
+            if (diagnostic.Properties.TryGetValue("_NewMappingForLocation", out var newMappingForLocation))
+            {
+                var baseTypeSyntax = root.FindToken(diagnostic.Location.SourceSpan.Start).Parent.AncestorsAndSelf().OfType<BaseTypeSyntax>().First();
+                var messageType = (baseTypeSyntax.Type as GenericNameSyntax).TypeArgumentList.Arguments[0];
+                var newMapping = SagaMessageMapping.CreateNewMapping(messageType, newMappingForLocation, correlationId);
+                mappings = mappings.Concat(new[] { newMapping });
+            }
+
+            // Register a code action that will invoke the fix
+            var action = CodeAction.Create(
+                title,
+                cancellationToken => RewriteConfigureHowToFindSagaMethod(context.Document, method, correlationId, mapperParamName, mappings.ToImmutableArray(), cancellationToken),
+                EquivalenceKey);
+
+            context.RegisterCodeFix(action, diagnostic);
+        }
+
+        static async Task<Document> RewriteConfigureHowToFindSagaMethod(
+            Document document,
+            MethodDeclarationSyntax method,
+            string correlationId,
+            string mapperParamName,
+            ImmutableArray<SagaMessageMapping> mappings,
+            CancellationToken cancellationToken)
+        {
+            // default(SyntaxNode).ToString() == empty string ""
+            var baseIndentText = method.GetLeadingTrivia().FirstOrDefault().ToString();
+            var indentText = baseIndentText.FirstOrDefault() == '\t' ? "\t" : "    ";
+
+            SyntaxTrivia Indent(int levels = 0)
+            {
+                // TODO: Probably a way to make this perform better
+                var indent = baseIndentText;
+                for (var i = 0; i < levels; i++)
+                {
+                    indent += indentText;
+                }
+                return Whitespace(indent);
+            }
+
+            var mapSagaInvocation = InvocationExpression(
+                MemberAccessExpression(
+                    SyntaxKind.SimpleMemberAccessExpression,
+                    IdentifierName(mapperParamName).WithLeadingTrivia(Indent(1)),
+                    IdentifierName("MapSaga")))
+                .WithArgumentList(CreateCorrelationIdMapping(correlationId));
+
+            for (int i = 0; i < mappings.Length; i++)
+            {
+                var mapping = mappings[i];
+
+                mapSagaInvocation = InvocationExpression(
+                    MemberAccessExpression(
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        mapSagaInvocation.WithTrailingTrivia(EndOfLine(NewLine)),
+                        Token(SyntaxKind.DotToken).WithLeadingTrivia(Indent(2)),
```

Comment:
Can we do something like method.GetLeadingTrivia() as an attempt to preserve the indentation from the user? Some could have spaces and other could have tabs.

Reply:
Never mind, looks like that is what you are determining in lines 95 and 96 to preserve the original indentation. Don't know if GetLeadingTriavia() is a "native" way to find out the same.

Reply:
Yep, trying to maintain the correct whitespace and indentation at least. But I need to indent more than just the original leading trivia.